ovirt_sriov.py

```diff
@@ -0,0 +1,384 @@
+               from the C(interface)." 
+        choices: ['all','specific']
+    networks:
+        description:
+            - "List of logical networks allowed to created VFs on the C(interface). If C(allowed_networks)
+               is set to I(specific), this setting is I(required)."
+    labels:
+        description:
+            - "List of labels to add to the specific C(networks). You I(cannot) add labels when 
+               C(allowed_networks) is set to I(all). C(labels) is ignored when C(allowed_networks)
+               is set to I(all)."
+
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Create 4 VFs on interface eth1 on host 'host1' in the example cluster/data center. 
+- name: Create VFs
+  ovirt_sriov:
+    name: example.host1
+    interface: eth1
+    vfs: 4
+    
+# Create 4 VFs with all networks allowed
+- name: Create VFs with specific networks
+  ovirt_sriov:
+    name: example.host1
+    interface: eth1
+    vfs: 4
+    allowed_networks: all
+  
+# Create 4 VFs on interface eth1 and only allow the net1 network to create VFs
+- name: Create VFs with specific networks
+  ovirt_sriov:
+    name: example.host1
+    interface: eth1
+    vfs: 4
+    allowed_networks: specific
+    networks:
+      - net1
+    
+# Everything in one
+- name: Create VFs
+  ovirt_sriov:
+    name: example.host1
+    interface: eth1
+    vfs: 4
+    allowed_networks: specific
+    networks:
+      - net1
+      - net2
+    labels:
+      - passive
+    
+'''
+
+RETURN = '''
+id:
+    description: ID of the SR-IOV host NIC managed.
+    returned: On success if host NIC is found.
+    type: str
+    sample: 7de90f31-222c-436c-a1ca-7e655bd5b60c
+sriov_config:
+    description: "Dictionary of all the host NIC VF attributes."
+    returned: On success if host NIC is found.
+    type: dict
+network_ids:
+    description: IDs of the networks allowed to create VFs from the interface. Will return None
+                 when allowed_networks is set to all.
+    returned: On success if host NIC is SR-IOV enabled.
+    type: list
+    sample: a0e60ec0-67a0-4209-a5d1-e7e563a061b5
+labels:
+    description: Label names given to the specific networks allowed to create VFs from the interface.
+                 Will return None if no labels were given, or allowed_networks is set to all.
+'''
+
+import traceback
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+from ansible.module_utils import six
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ovirt import (
+    BaseModule,
+    check_sdk,
+    create_connection,
+    equal,
+    get_dict_of_struct,
+    get_entity,
+    get_link_name,
+    ovirt_full_argument_spec,
+    search_by_name,
+)
+
+class SRIOVModule(BaseModule):
+
+    def build_entity(self):

```

Comment:
What is the use of this?  I do not see anywhere where it is called.  Is it junk code?

Reply:
It extends the Ansible base module - I never have to call build_entity, but I saw it written the same way in the official oVirt modules. Basically just copied what they had and it won't run without it. I guess I could scrap the module and just use functions, but then I'm passing a lot of values every time. 

I created a module object in main so I can reference all of the passed Ansible params without having to pass them to functions directly was my basic thought process.

Reply:
Ok, understood.  It's fine then.